rules/group.go

```diff
@@ -421,15 +424,21 @@ func (g *Group) CopyState(from *Group) {
 
 // Eval runs a single evaluation cycle in which all rules are evaluated sequentially.
 func (g *Group) Eval(ctx context.Context, ts time.Time) {
```

Comment:
Oof, how embarrassing. I didn't catch this because I was only testing a combination of sync & async rules in a group, and the sync rules depended on the async ones, so that acted as a waitgroup effectively.

I've addressed this in e75c48da134cd9b642cfc5be0221db3a2cfde9ce and added more tests.